### PR TITLE
skip module path segments

### PR DESCRIPTION
**Pull request by joaquin041**
Paths like this:
#[utoipauto(paths = "(./src/lib/bin/module from crate::bin)"]

Are getting resolved like:
crate::lib::bin::module::etc

And it should look like:
crate::bin::module::etc -> removing "lib" as we know that the module starts in "bin", same thing that is done with "src"

### CODE REVIEW

File: utoipauto-core/src/file_utils.rs [hunk 1]
```diff
@@ -81,19 +81,34 @@ pub fn extract_module_name_from_path(path: &str, crate_name: &str) -> String {
...
-    {
-        Some(idx) => &segments[(idx + 1)..],
-        None => &segments,
+    let segments_inside_crate = find_segment_and_skip(&segments, &["src", "tests"], 1);
+
+    // Also skip fragments that are already out of the crate name. For example,
+    // `./src/lib/my/module/name from crate::my::module` should turn into `crate::my::module:name`,
+    // and not into `crate::lib::my::module::name`.
+    let mut crate_segments = crate_name.split("::");
+    let first_crate_fragment = crate_segments.next().unwrap();
```
Comment (DenuxPlays) [inline utoipauto-core/src/file_utils.rs:90]: I am not a huge fan of unwraps.
Please replace it with expect and a descriptive error message

**Review by DenuxPlays — CHANGES_REQUESTED** [review]
Looks good 
Could you document the feature in the readme please

**Comment by DenuxPlays** [comment]
@ProbablyClem 
Can you also take a look at it.

The tests are my fault I am working on a pr to introduce a fix and some other feature.

File: utoipauto-core/src/file_utils.rs (hunk 1 above)
Comment (joaquin041) [reply utoipauto-core/src/file_utils.rs]: Done

**Review by DenuxPlays — COMMENTED** [review]
Looks good to me.

I would like if we can acceptance test this but I am not sure how to do this.

**Comment by DenuxPlays** [comment]
@joaquin041 

Couldn't we simply create a new folder in the `tests` folder that represents the issue you have and use this as an acceptance test?
Should be possible right?

**Comment by joaquin041** [comment]
I think that would do the trick, let me create another commit with that.

**Comment by joaquin041** [comment]
There it is. I am not sure if this will do. In order to properly test this, I needed to create another lib with a non-default "src" path. Let me know if this works.

**Review by DenuxPlays — CHANGES_REQUESTED** [review]
Looks good

Just a few minor notes

File: tests/crate_segment_path/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "utoipauto-lib-test"
+authors = ["joaquin041"]
+version = "0.0.1"
+edition = "2021"
```
[the remark this answers is not in the document]
Comment (joaquin041) [reply tests/crate_segment_path/Cargo.toml:5]: Done

**Review by DenuxPlays — APPROVED** [review]
Looks good to me.

Test passes (except for mine that I broke).

**Comment by DenuxPlays** [comment]
@joaquin041 Could you rebase this onto master to see if the chekcs no succeed

**Pull request merged**